src/main/java/org/jabref/logic/importer/fetcher/GoogleScholar.java

```diff
@@ -259,4 +230,46 @@ private void obtainAndModifyCookie() throws FetcherException {
             throw new FetcherException("Cookie configuration for Google Scholar failed.", e);
         }
     }
+
+    @Override
+    public Page<BibEntry> performSearchPaged(String query, int pageNumber) throws FetcherException {
+        LOGGER.debug("Using URL {}", query);
+        obtainAndModifyCookie();
+        List<BibEntry> foundEntries = new ArrayList<>(20);
+
+        URIBuilder uriBuilder = null;
+        try {
+            uriBuilder = new URIBuilder(BASIC_SEARCH_URL);
+        } catch (URISyntaxException e) {
+            throw new FetcherException("Error while fetching from " + getName() + " at URL " + BASIC_SEARCH_URL, e);
+        }
+
+        uriBuilder.addParameter("hl", "en");
+        uriBuilder.addParameter("start", String.valueOf(pageNumber * getPageSize()));
+        uriBuilder.addParameter("num", String.valueOf(getPageSize()));
+        uriBuilder.addParameter("btnG", "Search");
+        uriBuilder.addParameter("q", query);
+        String queryURL = uriBuilder.toString();
+
+        try {
+            addHitsFromQuery(foundEntries, queryURL);
+        } catch (IOException e) {
+            // if there are too much requests from the same IP address google is answering with a 503 and redirecting to a captcha challenge
+            // The caught IOException looks for example like this:
+            // java.io.IOException: Server returned HTTP response code: 503 for URL: https://ipv4.google.com/sorry/index?continue=https://scholar.google.com/scholar%3Fhl%3Den%26btnG%3DSearch%26q%3Dbpmn&hl=en&q=CGMSBI0NBDkYuqy9wAUiGQDxp4NLQCWbIEY1HjpH5zFJhv4ANPGdWj0
+            if (e.getMessage().contains("Server returned HTTP response code: 503 for URL")) {
```

Comment:
This struck me recently, isn't there a way to check if the exception is a more specific exception so that one could check for the status code number?

Reply:
I do not quite understand what the benefits of that would be. 
Could you provide me with an example to grasp your idea, please? :)